src/HDF5.jl

```diff
@@ -1712,10 +1712,31 @@ function create_external_dataset(parent::Union{File,Group}, name::AbstractString
     create_dataset(parent, name, datatype(t), dataspace(sz); dcpl=dcpl)
 end
 
+"""
+    do_write_chunk(dataset::Dataset, offset, chunk_bytes::Vector{UInt8}, filter_mask=0)
+
+    Write a raw chunk at a given offset.
+    offset is 1-based of rank `ndims(dataset)` and must fall on a chunk boundary.
+"""
 function do_write_chunk(dataset::Dataset, offset, chunk_bytes::Vector{UInt8}, filter_mask=0)
     checkvalid(dataset)
     offs = collect(hsize_t, reverse(offset)) .- 1
-    h5do_write_chunk(dataset, H5P_DEFAULT, UInt32(filter_mask), offs, length(chunk_bytes), chunk_bytes)
+    h5d_write_chunk(dataset, H5P_DEFAULT, UInt32(filter_mask), offs, length(chunk_bytes), chunk_bytes)
+end
+
+"""
+    do_read_chunk(dataset::Dataset, offset)
+
+    Read a raw chunk at a given offset.
+    offset is 1-based of rank `ndims(dataset)` and must fall on a chunk boundary.
+"""
+function do_read_chunk(dataset::Dataset, offset, filter_mask = Ref{UInt32}())
+    checkvalid(dataset)
+    offs = collect(hsize_t, reverse(offset)) .- 1
+    chunk_info = h5d_get_chunk_info_by_coord(dataset, offs)
+    chunk_bytes = Vector{UInt8}(undef, chunk_info[:size])
+    h5d_read_chunk(dataset, H5P_DEFAULT, offs, filter_mask, chunk_bytes)
```

Comment:
This method `h5d_read_chunk` could probably benefit from a more julian method as we have in `src/api_helpers.jl`  file.

Reply:
I added helpers for
* `h5d_get_chunk_storage_size`
* `h5d_get_num_chunks`
* `h5d_read_chunk`
* `h5d_write_chunk`

About `h5d_read_chunk`, I currently have it returning just the buffer and ignoring the filter mask. Should I also return the filter mask?